src/robustbase.py

```diff
@@ -40,117 +40,257 @@ def bias_corr(n):
     return corr_factor
 
 def median(x, low=False, high=False):
-    if low==True:
+    """
+    Calculate the median of a given list of values.
+
+    Parameters
+    ----------
+    x : list-like
+        List of numeric values.
+    low : bool, optional
+        If True, calculate the 'lo-median' (for even sample size, take the smaller of the two middle values).
+    high : bool, optional
+        If True, calculate the 'hi-median' (for even sample size, take the larger of the two middle values).
+
+    Returns
+    -------
+    median : float
+        The calculated median based on the provided options.
+
+    Notes
+    -----
+    The median is a measure of central tendency that represents the middle value in a sorted dataset.
+    It is robust to outliers and is often used as an alternative to the mean.
+
+    References
+    ----------
+    [1] https://docs.python.org/3/library/statistics.html#statistics.median
+    """
+
+    # Check if low flag is set
+    if low == True:
         return statistics.median_low(x)
-    
-    if high==True:
+
+    # Check if high flag is set
+    if high == True:
         return statistics.median_high(x)
-    
+
+    # Calculate the median using the default (high) method
```

Comment:
I've never seen this and this should definitely be mentioned in the docstring or changed.
@deepak7376 was this the intended behaviour? Because then the specification of `high` is
- redundant because this is the only possible outcome anyway if `low` is `False`
- misleading because it suggests that if `low` and `high` are `False`, the ordinary median is returned.